core/stream.go

```diff
@@ -232,6 +236,39 @@ func (c *MessageStream) Flush() error {
 	return nil
 }
 
+// Accept an SSLRequest and negotiate SSL using tlsConfig.  The underlying
+// ReadWriteCloser must implement net.Conn.
+func (c *MessageStream) AcceptSSL(tlsConfig *tls.Config) (connState tls.ConnectionState, err error) {
+	_, ok := c.rw.(net.Conn)
+	if !ok {
+		return connState, fmt.Errorf("ReadWriteCloser does not implement net.Conn")
+	}
+
+	err = c.SendSSLRequestResponse(AcceptSSLRequest)
+	if err != nil {
+		return connState, err
+	}
+
+	upgradeFunc := func (st io.ReadWriteCloser) (io.ReadWriteCloser, error) {
+		nc := st.(net.Conn)
+		uc := tls.Server(nc, tlsConfig)
+		err := uc.Handshake()
+		if err != nil {
+			return nil, err
+		}
+		connState = uc.ConnectionState()
+		return uc, nil
+	}
+	err = c.UpgradeConnection(upgradeFunc)
+	return connState, err
+}
+
+func (c *MessageStream) UpgradeConnection(up ConnectionUpgradeFunc) error {
+	var err error
```

Comment:
For what it's worth, I don't feel strongly one way or the other about minor style issues like that.